### PR TITLE
Refactor connecting p2sh into a separate method

### PR DESCRIPTION
**Pull request by pool2win**
This PR moves code that handles connecting p2sh into a separate function.

Let me know if these changes potentially break anything, I can add new tests to be sure we don't break anything.

### CODE REVIEW

**Comment by evoskuil** [comment]
Please rebase.

**Comment by pool2win** [comment]
> Please rebase.

Done.

**Review by evoskuil — APPROVED** [review]
The refactor makes methods smaller but also creates an additional stack call and input dereferences, but reasonable tradeoff for the improved readability.

File: include/bitcoin/system/impl/machine/interpreter.ipp [hunk 1]
```diff
@@ -1640,35 +1640,54 @@ connect(const context& state, const transaction& tx,
...
 
-        if (!program.is_true(false))
-            return error::stack_false;
+template <typename Stack>
+code interpreter<Stack>::connect_p2sh(const context& state,
+    const transaction& tx, const input_iterator& it,
+    interpreter& input_program) NOEXCEPT
+{
+    code ec;
+    const auto input = **it;
```
Comment (evoskuil) [inline include/bitcoin/system/impl/machine/interpreter.ipp:1662]: This is a copy of the input, should be:
`const auto input = **it;`

**Pull request merged**